### PR TITLE
De-duplicate architecture and concepts

### LINKED ISSUE #234: Overview: Architecture duplicates concepts
The "what is vitess" guide duplicates "Concepts" https://vitess.io/docs/overview/whatisvitess/

It is a long page, so Topology / vtgate / vttablet / vtctl / vtctld / vtworker can be removed from here and added to the concepts page where required.

### PR DESCRIPTION
**Pull request by morgo**
Fixes #234 

I think it could be improved further by switching out of the architecture diagram for a new one. But for now removing duplication helps with maintainability.

Signed-off-by: Morgan Tocker <tocker@gmail.com>

### CODE REVIEW

File: content/en/docs/concepts/vtctl.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: vtctl
+---
+
+**vtctl** is a command-line tool used to administer a Vitess cluster. It allows a human or application to easily interact with a Vitess implementation. Using vtctl, you can identify master and replica databases, create tables, initiate failovers, perform sharding (and resharding) operations, and so forth.
```
Comment (bnlandry) [inline content/en/docs/concepts/vtctl.md:5]: I think you can lose the parens around (and resharding). Readers tend to ignore anything in parens.

File: content/en/docs/concepts/vtworker.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+title: vtworker
+---
+
+**vtworker** hosts long-running processes. It supports a plugin architecture and offers libraries so that you can easily choose tablets to use. Plugins are available for the following types of jobs:
+
+* resharding differ jobs check data integrity during shard splits and joins
+* vertical split differ jobs check data integrity during vertical splits and joins
+
+vtworker also lets you easily add other validation procedures. You could do in-tablet integrity checks to verify foreign-key-like relationships or cross-shard integrity checks if, for example, an index table in one keyspace references data in another keyspace.
```
Comment (bnlandry) [inline content/en/docs/concepts/vtworker.md:10]: This sentence seems unclear. I would try breaking it up into two: You can do in-tablet... This is useful/applicable if...

**Review by bnlandry — APPROVED** [review]

File: content/en/docs/concepts/vtworker.md (hunk 2 above)
Comment (morgo) [reply content/en/docs/concepts/vtworker.md:10]: I will fix at a later point. This is an initial copy-paste from the architecture docs.

File: content/en/docs/concepts/vtctl.md (hunk 1 above)
Comment (morgo) [reply content/en/docs/concepts/vtctl.md:5]: I will fix at a later point. This is an initial copy-paste from the architecture docs.

**Pull request merged**